src/Features/ERDDAP/Platform/Observations/CurrentConditions/wind.tsx

```diff
@@ -1,81 +1,163 @@
+  if (windTimeSeries.length < 1) {
+    const noWindTimeSeries = platform.properties.readings.filter((timeSeries) =>
+      timeSeries.data_type.standard_name.includes("wind")
+    )
+
+    if (noWindTimeSeries.length < 1) {
+      return null
+    }
+
+    return <OtherWindCard platform={platform}>No recent wind data available.</OtherWindCard>
   }
 
-  const data: DataTimeSeries[] = datasets.map((dataset) => ({
-    name: dataset.data_type.long_name,
-    timeSeries: dataset.readings,
-    unit: dataset.data_type.units,
-  }))
+  if (timeSeries.length < 0) {
+    return <OtherWindCard platform={platform}>No wind data</OtherWindCard>
+  }
 
-  const speed = datasets.filter((dataset) => dataset.data_type.standard_name.includes("speed"))
-  const gust = datasets.filter((dataset) => dataset.data_type.standard_name.includes("gust"))
-  const direction = datasets.filter((dataset) => dataset.data_type.standard_name.includes("direction"))
+  return (
+    <UseDatasets
+      timeSeries={timeSeries}
+      loading={<OtherWindCard platform={platform}>Loading wind data</OtherWindCard>}
+      error={<OtherWindCard platform={platform}>Error loading wind data.</OtherWindCard>}
+    >
+      {({ datasets }) => {
+        const filteredDatasets = datasets
+          .map((ds) => ({
+            ...ds,
+            timeSeries: ds.timeSeries.filter((r) => aDayAgo < r.time),
+          }))
+          .filter((ds) => ds.timeSeries.length > 0)
 
-  const speedAndGust = [...speed, ...gust]
-  if (speedAndGust.length < 1) {
+        if (filteredDatasets.length > 0) {
+          return <DisplayWindCard {...{ platform, timeSeries, unit_system }} datasets={filteredDatasets} />
+        }
+        return <OtherWindCard platform={platform}>Error loading wind data.</OtherWindCard>
+      }}
+    </UseDatasets>
+  )
+}
+
+interface DisplayWindCardProps extends WindCardProps {
+  datasets: DataTimeSeries[]
+  unit_system: UnitSystem
+  timeSeries: PlatformTimeSeries[]
+}
+
+/**
+ * Display wind datasets on a current conditions card
+ */
+export const DisplayWindCard: React.FunctionComponent<DisplayWindCardProps> = ({ platform, datasets, unit_system }) => {
+  const { speed, gust, direction } = pickWindDatasets(platform, datasets)
+  const { speed: speedTimeSeries, gust: gustTimeSeries } = pickWindTimeSeries(platform)
+
+  if ((!speed && !gust) || !(speedTimeSeries || gustTimeSeries)) {
     return null
   }
 
-  const data_converter = converter(speedAndGust[0].data_type.standard_name)
+  const data_converter = converter(speedTimeSeries!.data_type.standard_name)
 
   let speedTitle: string = ""
-  if (speed.length > 0 && speed[0].readings.length > 0) {
+  if (speed) {
     speedTitle =
       " - " +
-      round(data_converter.convertToNumber(speed[0].readings[speed[0].readings.length - 1].reading, unit_system), 1) +
+      round(data_converter.convertToNumber(speed.timeSeries[speed.timeSeries.length - 1].reading, unit_system), 1) +
```

Comment:
Codacy found an issue: [Identifier 'data_converter' is not in camel case.](https://app.codacy.com/manual/gmri/Neracoos-1-Buoy-App/pullRequest?prid=6166252)